boto3/docs.py

```diff
@@ -591,11 +592,122 @@ def document_operation(operation_model, service_name, operation_name=None,
+                            key, value, indent=12, indent_first=False) +
+                          '\n' + param_desc)
         required = key in required_params and 'Required' or 'Optional'
         docs += ('      :param {0} {1}: *{2}* - {3}\n'.format(
-            param_type, key, required,
-            html_to_rst(value.documentation, indent=9)))
+            param_type, key, required, param_desc))
     if rtype is not None:
-        docs += '\n\n      :rtype: {0}\n\n'.format(rtype)
+        docs += '      :rtype: {0}\n\n'.format(rtype)
+
+    # Only document the return structure if it isn't a resource. Usually
+    # this means either a list or structure.
+    output_shape = operation_model.output_shape
+    if rtype in ['list', 'dict'] and output_shape is not None:
+        docs += ('      :return:\n         Structure description::\n\n' +
+                 document_structure(None, output_shape, indent=12) + '\n')
+
+    return docs
+
+
+PARAM_NAME = "'{name}': "
+ELLIPSIS = '...\n'
+STRUCT_START = '{\n'
+STRUCT_END = '}'
+LIST_START = '[\n'
+LIST_END = ']'
+
+
+def document_structure(name, shape, indent=0, indent_first=True,
+                       parent_type=None, eol='\n'):
+    """
+    Document a nested structure (list or dict) parameter or return value as
+    a snippet of Python code with dummy placeholders. For example:
+
+        {
+            'Param1': [
+                STRING,
+                ...
+            ],
+            'Param2': BOOLEAN,
+            'Param3': {
+                'Param4': FLOAT,
+                'Param5': INTEGER
+            }
+        }
+
+    :type name: string
+    :param name: The shape name.
+    :type shape: botocore.model.Shape
+    :param shape: The shape structure.
+    :type indent: integer
+    :param indent: The number of spaces to indent each line.
+    :type indent_first: boolean
+    :param indent_first: Whether to indent the first line.
+    :type parent_type: string
+    :param parent_type: The type name of the parent, which determines
+                        whether a name is printed for nested members.
+    :type eol: string
+    :param eol: The end-of-line string to use when finishing a member.
+    :rtype: string
+    """
+    docs = ''
+    spaces = ' ' * indent
+
+    # Add spaces if the first line is indented.
+    if indent_first:
+        docs += spaces
+
+    if shape.type_name == 'structure':
+        # Only include the name if the parent is also a structure.
+        if parent_type == 'structure':
+            docs += PARAM_NAME.format(name=name)
+
+        docs += STRUCT_START
+
+        # Go through each member and recursively process them.
+        for i, member_name in enumerate(shape.members):
+            # Individual members get a trailing comma only if they
+            # are not the last item.
+            member_eol = '\n'
+            if i < len(shape.members) - 1:
+                member_eol = ',' + member_eol
+
+            docs += document_structure(
```

Comment:
Nice catch. Everything seems to render fine but I'll update this to detect loops for the future.